include/nuttx/mm/map.h

```diff
@@ -162,6 +169,46 @@ void mm_map_destroy(FAR struct mm_map_s *mm);
  *
  ****************************************************************************/
 
+#ifdef CONFIG_ARCH_VMA_MAPPING
```

Comment:
should we emulate with kmm_malloc/umm_malloc in case of CONFIG_ARCH_VMA_MAPPING == n?

Reply:
Not in this place I think.

This only keeps track of virtual memory reservations with page granularity, so that it is easy to find a free virtual memory and find an address for a process to map memory to. And in theory this same info could be extracted from the list of mappings as well, it would just be a bit more complicated.

This info is not needed at all when you don't have virtual memory/MMU, it is easier to just return the physical address in mmap, as what is done already in rammap and anonmap (and what we do also in shmfs for flat&protected modes).

So, IMHO, there is nothing to re-use in here for rammap and anonmap.